### PR TITLE
refactoring_parser_server_worker

### PR DESCRIPTION
**Pull request by amveng**

### CODE REVIEW

File: app/workers/parser_site_worker.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    sites.each do |site|
+      parsing_site(site, search_patterns)
+    end
+  end
+
+  def parsing_site(site, search_patterns)
+    document = Services::ParserEngine::GettingDataSite.new(site.url).take
+    return if document.blank?
+
+    selected_data = Services::ParserEngine::DataPreparation.new(document, site.css_selector).selected
```
Comment (annvelents) [inline app/workers/parser_site_worker.rb:18]: what do you think about not DataPreparation, but DataSelector?

File: app/workers/parser_site_worker.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    array_of_match_numbers = Services::ParserEngine::SearchByPatterns.new(selected_data, search_patterns).collect_array
+    array_of_new_servers = Services::ParserEngine::MakingServersFromData.new(array_of_match_numbers, selected_data, site).make
```
Comment (annvelents) [inline app/workers/parser_site_worker.rb:20]: this can be ServersBuilder

File: app/workers/parser_site_worker.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+  def perform
+    search_patterns = Serverversion.pluck(:name)
+    sites = ParserSite.where(enabled: true)
+    sites.each do |site|
+      parsing_site(site, search_patterns)
+    end
+  end
+
+  def parsing_site(site, search_patterns)
```
Comment (annvelents) [inline app/workers/parser_site_worker.rb:14]: also this should be parse_site

File: app/workers/parser_site_worker.rb (hunk 1 above)
Comment (annvelents) [reply app/workers/parser_site_worker.rb:18]: and method will be select

File: app/workers/parser_site_worker.rb (hunk 2 above)
Comment (annvelents) [reply app/workers/parser_site_worker.rb:20]: Also I think if we can add one more service that will combine all these functions... and is it needed

File: app/workers/parser_site_worker.rb [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    document = Services::ParserEngine::GettingDataSite.new(site.url).take
+    return if document.blank?
+
+    selected_data = Services::ParserEngine::DataPreparation.new(document, site.css_selector).selected
+    array_of_match_numbers = Services::ParserEngine::SearchByPatterns.new(selected_data, search_patterns).collect_array
+    array_of_new_servers = Services::ParserEngine::MakingServersFromData.new(array_of_match_numbers, selected_data, site).make
+    Services::ParserEngine::SaveServersAndCreateReport.new(array_of_new_servers, site.url).finishing_touch
```
Comment (annvelents) [inline app/workers/parser_site_worker.rb:21]: Existence "and" in the name of service means that this service does two action, when it should be one :)

File: app/workers/parser_site_worker.rb [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  def perform
+    search_patterns = Serverversion.pluck(:name)
+    sites = ParserSite.where(enabled: true)
+    sites.each do |site|
+      parsing_site(site, search_patterns)
+    end
+  end
+
+  def parsing_site(site, search_patterns)
+    document = Services::ParserEngine::GettingDataSite.new(site.url).take
```
[the remark this answers is not in the document]
Comment (amveng) [reply app/workers/parser_site_worker.rb:15]: maybe then SiteGraber?

**Pull request merged**